### PR TITLE
CHE-249: Setting workspace Pod terminationGracePeriodSeconds to zero

### PR DESCRIPTION
**Pull request by ibuziuk**
### What does this PR do?
Setting workspace Pod terminationGracePeriodSeconds to zero

### What issues does this PR fix or reference?
https://issues.jboss.org/browse/CHE-249

### CODE REVIEW

File: plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftConnector.java [hunk 1]
```diff
@@ -1530,6 +1532,7 @@ private String waitAndRetrieveContainerID(String deploymentName) throws IOExcept
                 if (OPENSHIFT_POD_STATUS_RUNNING.equals(status)) {
                     String containerID = pod.getStatus().getContainerStatuses().get(0).getContainerID();
                     String normalizedID = KubernetesStringUtils.normalizeContainerID(containerID);
+                    pod.getSpec().setTerminationGracePeriodSeconds(OPENSHIFT_POD_TERMINATION_GRACE_PERIOD);
```
Comment (amisevsk) [inline plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftConnector.java]: Doing it this way will not update the actual Pod spec. I think we need to add this in `createOpenShiftDeployment()`

```
        PodSpec podSpec = new PodSpecBuilder()
                                 .withContainers(container)
                                 .withVolumes(getVolumesFrom(volumes, workspaceID))
                                 .withTerminationGracePeriodSeconds(OPENSHIFT_POD_TERMINATION_GRACE_PERIOD)
                                 .build();
```

**Review by amisevsk — CHANGES_REQUESTED** [review]
Thanks Ilya! Looks mostly good -- just a minor issue with how the grace period is being set. 

I've tested shutting down a workspace with the 0 second grace period, and it brings workspace shutdown down to ~5 seconds when running on minishift.

**Comment by ibuziuk** [comment]
@amisevsk oh, thank you for review - I had no chance to test this PR due to weird issue with building che (I even posted a question on stack overflow - http://stackoverflow.com/questions/43792427/maven-hangs-forever-during-the-project-build-used-to-work-fine) lemme update the PR

**Comment by ibuziuk** [comment]
PR has been updated

**Review by amisevsk — APPROVED** [review]
LGTM! Thanks.

**Review by l0rd — APPROVED** [review]
LGTM great job @ibuziuk and @amisevsk

**Comment by ibuziuk** [comment]
@l0rd @amisevsk  shall we apply ? (code freeze is over I guess)

**Comment by l0rd** [comment]
Yes go ahead

Il 05 mag 2017 00:30, "Ilya Buziuk" <notifications@github.com> ha scritto:

**Pull request merged**